regression_tests/operator.go

```diff
@@ -0,0 +1,262 @@
+		fmt.Println("error running tests:", err)
+	}
+}
+
+// handleUserCreation handles the creation of a user.
+func handleUserCreation(paymailAlias string, config *regressionTestConfig) (*regressionTestUser, error) {
+	if config.ClientOneLeaderXPriv != "" {
+		if promptUserAndCheck("Would you like to use user from env? (y/yes or n/no): ") == 1 {
+			return useUserFromEnv(config, paymailAlias)
+		}
+	}
+
+	user, err := createUser(paymailAlias, config)
+	if err != nil {
+		return handleCreateUserError(err, paymailAlias, config)
+	}
+
+	return user, nil
+}
+
+// handleCreateUserError handles the error when creating a user.
+func handleCreateUserError(err error, paymailAlias string, config *regressionTestConfig) (*regressionTestUser, error) {
+	if err.Error() == "paymail already exists" {
+		return handleExistingPaymail(paymailAlias, config)
+	} else {
+		return nil, fmt.Errorf("error creating user: %v", err)
+	}
+}
+
+// handleExistingPaymail handles the case when the paymail already exists.
+func handleExistingPaymail(paymailAlias string, config *regressionTestConfig) (*regressionTestUser, error) {
+	if promptUserAndCheck("Paymail already exists. Would you like to use it (you need to have xpriv)? (y/yes or n/no): ") == 1 {
+		return useExistingPaymail(paymailAlias, config)
+	}
+	if promptUserAndCheck("Would you like to recreate user? (y/yes or n/no):") == 1 {
+		return recreateUser(paymailAlias, config)
+	}
+	return nil, fmt.Errorf("user should be recreated or xpriv should be provided")
+}
+
+// recreateUser deletes and recreates the user.
+func recreateUser(paymailAlias string, config *regressionTestConfig) (*regressionTestUser, error) {
+	err := deleteUser(paymailAlias, config)
+	if err != nil {
+		return nil, fmt.Errorf("error deleting user: %v", err)
+	}
+	user, err := createUser(paymailAlias, config)
+	if err != nil {
+		return nil, fmt.Errorf("error creating user after deletion: %v", err)
+	}
+	return user, nil
+}
+
+// useExistingPaymail uses an existing paymail address.
+func useExistingPaymail(paymailAlias string, config *regressionTestConfig) (*regressionTestUser, error) {
+	validatedXPriv := getValidXPriv()
+	keys, err := xpriv.FromString(validatedXPriv)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing xpriv: %v", err)
+	}
+	return &regressionTestUser{
+		XPriv:   keys.XPriv(),
+		XPub:    keys.XPub().String(),
+		Paymail: preparePaymail(paymailAlias, config.ClientOneURL),
+	}, nil
+}
+
+// handleFundsTransfer handles the transfer of funds to a user.
+func handleFundsTransfer(user *regressionTestUser, config *regressionTestConfig) error {
+	response := promptUserAndCheck("Do you have xpriv and master instance URL? (y/yes or n/no): ")
+	if response == 0 {
+		fmt.Printf("Please send %d Sato for full regression tests:\n%s\n", minimalBalance, user.Paymail)
+		isSent := 0
+		for isSent < 1 {
+			isSent = promptUserAndCheck("Did you make the transaction? (y/yes or n/no): ")
+		}
```

Comment:
Maybe we should define a way to cancel and/or add a timeout?

Reply:
timeout added